Source/Project64/User Interface/Frame Per Second Class.cpp

```diff
@@ -102,79 +102,129 @@ void CFramePerSecond::DisplayViCounter(DWORD FrameRate)
 	if (m_iFrameRateType == FR_PERCENT)
 	{
 		float Percent;
-		if (FrameRate != 0) 
-        {
+		if (FrameRate != 0)
+		{
 			Percent = ((float)FrameRate) / m_ScreenHertz;
 		}
-        else 
-        {
-			if (CurrentFrame > (NoOfFrames << 3)) 
-            {
+		else
+		{
+			if (CurrentFrame > (NoOfFrames << 3))
+			{
 				__int64 Total;
-				
+
 				Total = 0;
-				for (int count = 0; count < NoOfFrames; count ++) 
-                {
+				for (int count = 0; count < NoOfFrames; count++)
+				{
 					Total += Frames[count];
 				}
-				Percent = ((float)(Frequency/ ((double)Total / (NoOfFrames << 3)))) / m_ScreenHertz;
+				Percent = ((float)(Frequency / ((double)Total / (NoOfFrames << 3)))) / m_ScreenHertz;
 			}
-            else 
-            {
+			else
+			{
 				g_Notify->DisplayMessage2(L"");
 				return;
 			}
 		}
-		g_Notify->DisplayMessage2(L"%.1f %%",Percent * 100);
+		g_Notify->DisplayMessage2(L"%.1f %%", Percent * 100);
 	}
-}
+	if (m_iFrameRateType == FR_VIs_DLs) //added new selection show DL/s and VI/s
+	{
+		if (FrameRate != 0)
+		{
+			g_Notify->DisplayMessage2(L"VI/s: %d.00", FrameRate);
+		}
+		else
+		{
+			if (CurrentFrame > (NoOfFrames << 3))
+			{
+				__int64 Total;
 
-void CFramePerSecond::FrameRateTypeChanged (CFramePerSecond * _this)
-{
-	_this->m_iFrameRateType    = g_Settings->LoadDword(UserInterface_FrameDisplayType);
-	_this->Reset(true);
-}
+				Total = 0;
+				for (int count = 0; count < NoOfFrames; count++)
+				{
+					Total += Frames[count];
+				}
+				g_Notify->DisplayMessage2(L"VI/s: %.2f", Frequency / ((double)Total / (NoOfFrames << 3)));

```

Comment:
is the display here just the same as FR_VIs or is the DL count meant to be here as well